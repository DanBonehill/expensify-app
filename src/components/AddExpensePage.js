import React from "react";

const AddExpensePage = () => (
    <div>
        This is from my Add Expense
    </div>
);

export default AddExpensePage