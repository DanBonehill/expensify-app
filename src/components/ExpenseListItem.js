import React from 'react';

export default ({description, amount, createdAt}) => (
    <div>
        <ul>
            <li>{description}</li>
            <li>{amount}</li>
            <li>{createdAt}</li>
        </ul>
    </div>
)