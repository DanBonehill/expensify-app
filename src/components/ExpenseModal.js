import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement('#app');

const ExpenseModal = (props) => {
    return (
        <Modal
            isOpen={props.isOpen}
            onRequestClose={props.closeModal}
            closeTimeoutMS={150}
            ariaHideApp={ !process.env.NODE_ENV === 'test' }
            className="modal"
        >
            <h3 className="modal__title">Are you sure?</h3>
            <div>
                <button onClick={props.onDeleteClick} className="button button--delete">Remove</button>
                <button onClick={props.closeModal} className="button button--link">Cancel</button>
            </div>
        </Modal>
    )
};

export default ExpenseModal