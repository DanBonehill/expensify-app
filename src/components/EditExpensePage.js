import React from "react";
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startEditExpense, startRemoveExpense} from "../actions/expenses";
import ExpenseModal from './ExpenseModal';

export class EditExpensePage extends React.Component {
    state = {
        isOpen: false
    };
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onClick = () => {
        this.setState(() => ({
            isOpen: true
        }));
    };
    onDeleteClick = () => {
        this.props.startRemoveExpense({id: this.props.expense.id});
        this.props.history.push('/');
    };
    closeModal = () => {
        this.setState(() => ({
            isOpen: false
        }))
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onClick}>Remove Expense</button>
                    <ExpenseModal
                        isOpen={this.state.isOpen}
                        closeModal={this.closeModal}
                        onDeleteClick={this.onDeleteClick}
                    />
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)