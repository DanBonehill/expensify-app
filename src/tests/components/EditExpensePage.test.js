import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import {EditExpensePage} from '../../components/EditExpensePage';

describe('EditExpensePage snapshot', () => {
    const editExpense = jest.fn();
    const removeExpense = jest.fn();
    const history = {push: jest.fn()};
    const wrapper = shallow(
        <EditExpensePage
            editExpense={editExpense}
            removeExpense={removeExpense}
            history={history}
            expense={expenses[0]}
        />
    );

    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should handle editExpense', () => {
        wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
        expect(history.push).toHaveBeenLastCalledWith('/');
        expect(editExpense).toHaveBeenLastCalledWith(expenses[0].id, expenses[0]);
    });

    test('should handle removeExpense', () => {
        wrapper.find('button').simulate('click');
        expect(history.push).toHaveBeenLastCalledWith('/');
        expect(removeExpense).toHaveBeenLastCalledWith({id: expenses[0].id});
    })
});