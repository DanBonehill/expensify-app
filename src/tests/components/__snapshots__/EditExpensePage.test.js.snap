// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditExpensePage snapshot should render correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 10500,
        "createdAt": 0,
        "description": "Rent",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
